src/datasets/utils/file_utils.py

```diff
@@ -177,6 +178,17 @@ def cached_path(
     if isinstance(url_or_filename, Path):
         url_or_filename = str(url_or_filename)
 
+    # Convert fsspec URL in the format "file://local/path" to "local/path"
+    # Copied from: https://github.com/fsspec/filesystem_spec/blob/14a778879b6a68b65031ef86e232b9d88c005bab/fsspec/implementations/local.py#L214-L221
+    if url_or_filename.startswith("file://"):
+        url_or_filename = url_or_filename[7:]
+    elif url_or_filename.startswith("file:"):
+        url_or_filename = url_or_filename[5:]
+    elif url_or_filename.startswith("local://"):
+        url_or_filename = url_or_filename[8:]
+    elif url_or_filename.startswith("local:"):
+        url_or_filename = url_or_filename[6:]
```

Comment:
Maybe use LocalFileSystem._strip_protocol instead ?

Reply:
`LocalFileSystem._strip_protocol` always returns an absolute path, but I think this is fine for `cached_path` (there was a test checking relative paths are preserved, so I had to update it)